src/pages/orders/Order.tsx

```diff
@@ -5,27 +5,33 @@ import type { TableProps } from 'antd/es/table';
 import { BaseButton } from 'components';
 import { BaseOrderStatus } from 'components/OrderStatus';
 import dayjs from 'dayjs';
-import { ReactElement, useMemo } from 'react';
+import { ReactElement, useEffect, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useNavigate } from 'react-router-dom';
 import { routePaths } from 'routes/routeComponent';
 import { THC } from 'utils/constants';
-import { getExactPath } from 'utils/getExactPath';
 import { OrderStyled } from './Order.styles';
-import { OrderProps } from './withOrderController';
-interface DataType extends OperationalSetting {
-  key: string;
-  name: string;
-  createdAt: Date;
+import { OperationalSettingDrawer } from './drawer';
+import { IDrawerData } from './drawer/withDrawerController';
+import { IOperationalSettingProps } from './withOrderController';
+export interface IOrderGroup extends OperationalSetting {
   orders?: Order[];
 }
 
-export const Orders = (props: OrderProps): ReactElement => {
+export const Orders = (props: IOperationalSettingProps): ReactElement => {
   const { data, dispatch } = props;
-  const { isLoading, isStatusLoading, groupOrders, operationalSettings } = data;
+  const { isLoading, statusLoading, groupOrders, operationalSettings } = data;
+  const { handleOnChangeOrderStatus } = dispatch;
   const navigate = useNavigate();
   const { t } = useTranslation('order');
 
+  const [drawerData, setDrawerData] = useState<Partial<IDrawerData>>({ isOpen: false, statusLoading });
+  console.log(drawerData);
+
+  useEffect(() => {
```

Comment:
I think it has been improved.

`
useEffect(() => {
  setDrawerData((prevDrawerData) => {
    const updatedStatus = operationalSettings?.[prevDrawerData?._id || '']?.status;
    return {
      ...prevDrawerData,
      statusLoading,
      status: !!updatedStatus  ? updatedStatus : prevDrawerData.status,
    };
  });
}, [statusLoading, operationalSettings]);
`